### ISSUE TITLE
Document and clarify that index_cols does not apply to Image elements

### ISSUE THREAD

**Issue by LaurensvanG** [holoviz/holoviews#5118]
ALL software version info
(this library, plus any other relevant software, e.g. bokeh, python, notebook, OS, browser, etc)
holoviews: 1.14.5
bokeh: 2.3.3
Python: 3.8.10
OS: Windows 11
Browser: Chrome

#### Description of expected behavior and the observed behavior
Using the `index_cols` option on a `hv.link_selections.instance` fails on a `hv.Image` element when making a selection, using both `box_select` and `lasso_select`.

#### Complete, minimal, self-contained example code that reproduces the issue

```Python
import numpy as np, xarray as xr, holoviews as hv
hv.extension("bokeh")

data = np.random.rand(100, 50)
da = xr.Dataset(
    {
        "data": (("x", "y"), data)
    },
    {
        "x": np.arange(data.shape[0]),
        "y": np.arange(data.shape[1]),
    }
)
ds = hv.Dataset(da, ["x", "y"])
ls = hv.link_selections.instance(index_cols=["x", "y"])
ls(hv.Image(ds, ["x", "y"], "data"))

```

#### Stack traceback and/or browser JavaScript console output
Python failed with the following traceback: 
C:\Users\gieri\miniconda3\envs\dashboard\lib\site-packages\pyviz_comms\__init__.py _handle_msg L325
C:\Users\gieri\miniconda3\envs\dashboard\lib\site-packages\holoviews\plotting\bokeh\callbacks.py on_msg L169
C:\Users\gieri\miniconda3\envs\dashboard\lib\site-packages\holoviews\plotting\bokeh\callbacks.py on_msg L161
C:\Users\gieri\miniconda3\envs\dashboard\lib\site-packages\holoviews\streams.py trigger L186
C:\Users\gieri\miniconda3\envs\dashboard\lib\site-packages\holoviews\streams.py perform_update L877
C:\Users\gieri\miniconda3\envs\dashboard\lib\site-packages\holoviews\streams.py event L428
C:\Users\gieri\miniconda3\envs\dashboard\lib\site-packages\holoviews\streams.py update L442
C:\Users\gieri\miniconda3\envs\dashboard\lib\site-packages\holoviews\streams.py transform L1038
C:\Users\gieri\miniconda3\envs\dashboard\lib\site-packages\holoviews\streams.py transform L911
C:\Users\gieri\miniconda3\envs\dashboard\lib\site-packages\holoviews\streams.py transform_function L1061
C:\Users\gieri\miniconda3\envs\dashboard\lib\site-packages\holoviews\element\selection.py _get_selection_expr_for_stream_value L277
C:\Users\gieri\miniconda3\envs\dashboard\lib\site-packages\holoviews\element\selection.py _skip L258
	AttributeError: 'Image' object has no attribute '_index_skip'

**Comment by jbednar** [comment]
I can reproduce this result, but is it a bug?  I can't tell if `index_cols` were ever intended to be used for a 2D-indexed structure like an image. The docs for it are:

> If provided, selection switches to index mode where all queries are expressed solely in terms of discrete values along the index_cols.  All Elements given to link_selections must define the index_cols, either as explicit dimensions or by sharing an underlying Dataset that defines them.

Seems ambiguous, given the plural form `cols`. @philippjfr ?

**Comment by jbednar** [comment]
Good point! Yes, an Image has 2 index dimensions, but only one of the two dimensions is a column, so you're right that the nomenclature does already explicitly exclude Images.  I'll rename this issue as a request for documenting that Images aren't covered by `index_cols`.

**Comment by PeterLindner** [comment]
Hi @jbednar , @philippjfr 

sorry for reviving this old issue but this is a problem for us when using datashader. Consider the following modified code from the [linked brushing guide](https://holoviews.org/user_guide/Linked_Brushing.html)

```python
import holoviews as hv
import panel as pn
from bokeh.sampledata.autompg import autompg
from holoviews.operation.datashader import dynspread, rasterize
from holoviews.selection import link_selections

hv.extension("bokeh")

autompg_ds = hv.Dataset(autompg, ['yr', 'name', 'origin'])

w_accel_scatter = hv.Points(autompg_ds, ['weight', 'accel'])
w_accel_scatter_rasterized = dynspread(rasterize(w_accel_scatter))

mpg_lsp = link_selections.instance(index_cols=['yr', 'origin'])

pn.Row(mpg_lsp(w_accel_scatter + w_accel_scatter_rasterized)).servable()
```

selecting on the normal `Points` plot correctly selects all points with the same `yr` and `origin` (`name` was intentionally excluded), but selecting on the rasterized plot fails with the error above.

I really hope you reconsider supporting index based selection for `Image`s, because we want to link two different datasets, that cannot easily be combined into one (ie range based selection is not feasible), and have to use datashader due to data volume.

**Comment by jbednar** [comment]
@PeterLindner, please open a separate issue so that we can address it, as datashaded items are different from items that start as Images.
